pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -34,8 +38,70 @@ func (*AgentConfig) Dependencies() []asset.Asset {
 	return []asset.Asset{}
 }
 
-// Generate generates the Agent manifest.
+// Generate generates the Agent Config manifest.
 func (a *AgentConfig) Generate(dependencies asset.Parents) error {
+
+	// TODO: We are temporarily generating a template of the agent-config.yaml
+	// Change this when its interactive survey is implemented.
+	agentConfigTemplate := `#
+# Note: This is an sample AgentConfig file showing
+# which fields are available to aid you in creating your 
+# own agent-config.yaml file.
+#
+apiVersion: v1
+kind: AgentConfig
+metadata:
+  name: example-agent-config
+  namespace: cluster0
+spec:
+# All fields are optional
+  rendezvousIP: your-node0-ip
+  hosts:
+  # If a host is listed, then at least one interface
+  # needs to be specified.
+  - hostname: change-to-hostname
+    role: master
+    # For more information about rootDeviceHints:
+    # https://docs.openshift.com/container-platform/4.10/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#root-device-hints_ipi-install-installation-workflow
+    rootDeviceHints:
+	  deviceName: /dev/sda
+    interfaces:
+      - macAddress: 00:00:00:00:00:00
+        name: host-network-interface-name
+    # networkConfig contains the NMStateConfig.Spec.Config.interfaces definition for this host.
+    # https://nmstate.io/examples.html
+    # NMStateConfig.Spec.Interfaces section is not included in networkConfig. It is specified
+    # in the above interfaces: node
```

Comment:
I feel like most people interacting with this file won't be familiar with the NMStateConfig resource at all. I'd be tempted to reduce this to just:

```
# networkConfig contains the network configuration for the host in NMState format.
# See https://nmstate.io/examples.html for examples.
```